library/build.gradle

```diff
@@ -15,32 +15,92 @@ apply plugin: 'signing'
 archivesBaseName = 'android-maps-utils'
 group = 'com.google.maps.android'
 
-dependencies {
-    compile 'com.google.android.gms:play-services-maps:6.5.87'
-}
-
-
 android {
     compileSdkVersion 19
     buildToolsVersion "21.1.2"
+    defaultPublishConfig "release"
+    publishNonDefault true
 
     sourceSets {
         main {
             manifest.srcFile 'AndroidManifest.xml'
+        }
+
+        gms{
             java.srcDirs = ['src']
             res.srcDirs = ['res']
         }
 
+        m4b {
+            java.srcDirs = ['m4b/src']
+            res.srcDirs = ['m4b/res']
+        }
+
         androidTest {
             java.srcDirs = ['tests/src']
         }
     }
+
+    productFlavors {
+        gms {
+            dependencies {
+                compile 'com.google.android.gms:play-services-maps:6.5.87'
+            }
+        }
+
+        m4b {
+            dependencies {
+                compile files('../maps_m4b.jar')
+            }
+        }
+    }
 }
 
+task buildM4b (dependsOn: build)
```

Comment:
Yes, but genM4bSrc and genM4bRes were kicked by the dependencies of preBuild task.
I'll just remove this task since we can run M4B tasks every time.